### PR TITLE
US012 - Add a new role

### PR DESCRIPTION
**Pull request by ManpreetM01**

### CODE REVIEW

**Comment by ManpreetM01** [comment]
Screenshot 2023-09-21 at 12 10 03

**Comment by ManpreetM01** [comment]
Screenshot 2023-09-26 at 09 04 36

**Comment by ManpreetM01** [comment]
Screenshot 2023-09-26 at 09 06 37

**Comment by ManpreetM01** [comment]
Screenshot 2023-09-26 at 09 07 29

**Comment by ManpreetM01** [comment]
Screenshot 2023-09-26 at 09 07 54

**Comment by ManpreetM01** [comment]
Changed 'JobSpecSummary' to 'Job Specification Summary' on create page


Screenshot 2023-09-26 at 09 15 04

File: src/controller/jobRoleController.ts [hunk 1]
```diff
@@ -52,4 +52,36 @@ export namespace JobRoles {
             });
         }
     }
+
+    export async function getCreate(req:Request, res:Response): Promise<void> {
+        const bands: JobBand[] = await bandService.getBands();
+        const capabilities: JobCapability[] = await capabilityService.getCapabilities();
```
Comment (TomBillingtonUK) [inline src/controller/jobRoleController.ts]: What is the user experince if any of the services throws an exception?

File: src/controller/jobRoleController.ts [hunk 2]
```diff
@@ -52,4 +52,36 @@ export namespace JobRoles {
...
+
+        res.render("create-job-role", {bands, capabilities});
+    }
+
+    export async function create(req: Request, res: Response): Promise<void> {
```
Comment (TomBillingtonUK) [inline src/controller/jobRoleController.ts]: `postCreate` to stay with the current naming convention

File: src/controller/jobRoleController.ts [hunk 3]
```diff
@@ -52,4 +52,36 @@ export namespace JobRoles {
...
+        try {
+            const jobRoleToCreate: JobRoleToUpdate = {
+                jobRoleName: jobRoleName,
+                jobSpecSummary:jobSpecSummary ,
+                band: band,
+                capability: capability,
+                responsibilities: responsibilities,
+                sharePoint: sharePoint
+            }
+            id = await jobRoleService.createJobRole(jobRoleToCreate)
```
Comment (TomBillingtonUK) [inline src/controller/jobRoleController.ts]: You are submitting unsanitised user input directly to the API. Input we know may fail/cause issues

File: src/service/jobRoleService.ts [hunk 4]
```diff
@@ -23,5 +23,17 @@ export const jobRoleService = {
...
+    async createJobRole(jobRole: JobRoleToCreate): Promise<number> {
+        try {
+            const response = await axios.post(`${process.env.API_URL}api/job-roles/`, jobRole)
+            if (response.status === 200) {
+                return response.data;
+            }
+            throw new Error('Create failed');
+        } catch (e) {
+            throw new Error(e.response.data);
+            //throw new Error('Failed to create job role');
```
Comment (TomBillingtonUK) [inline src/service/jobRoleService.ts]: Comment

File: views/job-roles.html [hunk 5]
```diff
@@ -24,4 +24,5 @@
             </tr>
         {% endfor %}
     </table>
+    <a href="/create-job-role" class="white btn btn-primary">Create</a>
```
Comment (TomBillingtonUK) [inline views/job-roles.html]: Can you add a little bit of space between this button and the table? Also if the table is really long then this button won't be on the screen so maybe positioning it at the top would be better

File: src/controller/jobRoleController.ts (hunk 1 above)
Comment (ManpreetM01) [reply src/controller/jobRoleController.ts]: Screenshot 2023-09-26 at 13 37 26

File: src/controller/jobRoleController.ts (hunk 3 above)
Comment (ManpreetM01) [reply src/controller/jobRoleController.ts]: changes made

**Review by TomBillingtonUK — APPROVED** [review]

**Review by RobertCrabtreeKainos — APPROVED** [review]

**Pull request merged**